src/main/java/cope/cosmos/client/features/modules/visual/BreadcrumbsModule.java

```diff
@@ -0,0 +1,132 @@
+package cope.cosmos.client.features.modules.visual;
+
+import cope.cosmos.client.features.modules.Category;
+import cope.cosmos.client.features.modules.Module;
+import cope.cosmos.client.features.setting.Setting;
+import cope.cosmos.util.string.ColorUtil;
+import net.minecraft.util.math.Vec3d;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+
+import static org.lwjgl.opengl.GL11.*;
+
+/**
+ * @author Wolfsurge
+ * @since 01/04/22
+ */
+public class BreadcrumbsModule extends Module {
+
+    // Settings
+    public static final Setting<Boolean> infinite = new Setting<>("Infinite", false).setDescription("Makes breadcrumbs last forever");
+    public static final Setting<Float> lifespanValue = new Setting<>("Lifespan", 10F, 100F, 1000F, 0).setDescription("The lifespan of the positions in ticks").setVisible(() -> !infinite.getValue());
+    public static final Setting<Float> lineWidth = new Setting<>("LineWidth", 0.1F, 3F, 5F, 1).setDescription("The width of the lines");
+
+    // List of positions
+    private final ArrayList<Position> positions = new ArrayList<>();
+
+    public BreadcrumbsModule() {
+        super("Breadcrumbs", Category.VISUAL, "Draws a trail behind you");
+    }
+
+    @Override
+    public void onEnable() {
+        // Clear positions
+        positions.clear();
+    }
+
+    @Override
+    public void onDisable() {
+        // Clear positions
+        positions.clear();
+    }
+
+    @Override
+    public void onTick() {
+        if (!nullCheck() || mc.player.ticksExisted <= 20) {
```

Comment:
pretty sure there's an event for this, iirc it was like ConnectEvent or something